### PR TITLE
UCP: Update RMA/AMO lane selection logic.

### LINKED ISSUE #1115: error: up to 8 memory domains are supported
The following error is printed in case there are more than 8 memory domains for usage on a host:
ucp_context.c:535  UCX  ERROR Only up to 8 memory domains are supported (have: 10)

As a workaround, can specify the UCX_TLS parameter or the UCX_NET_DEVICES parameter from the command line.
Can be reproduced on orion with a simple command line that uses '-mca pml ucx'

According to @yosefe : 
should be solved by adding flag to mm that it does not require registration

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1115

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1082/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3031/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls take a look

File: src/ucp/core/ucp_ep.inl [hunk 1]
```diff
@@ -85,11 +85,11 @@ static inline uct_md_h ucp_ep_md(ucp_ep_h ep, ucp_lane_index_t lane)
...
-static inline ucp_md_map_t ucp_lane_map_get_lane(ucp_md_lane_map_t lane_map,
-                                                 ucp_lane_index_t lane)
-{
-    return (lane_map >> (lane * UCP_MD_INDEX_BITS)) & UCS_MASK(UCP_MD_INDEX_BITS);
-}
+//static inline ucp_md_map_t ucp_lane_map_get_lane(ucp_md_lane_map_t lane_map,
+//                                                 ucp_lane_index_t lane)
+//{
+//    return (lane_map >> (lane * UCP_MD_INDEX_BITS)) & UCS_MASK(UCP_MD_INDEX_BITS);
+//}
```
Comment (shamisp) [inline src/ucp/core/ucp_ep.inl]: Looks like a leftover from a debug session

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1091/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3032/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1102/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3038/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3040/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3041/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3042/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1112/ for details.

**Pull request merged**